test_scripts/iAP2_Emulation/iAP2_Bluetooth_to_USB_transport_switch_happy_path.lua

```diff
@@ -0,0 +1,158 @@
+local iAP2_BT_Port = 23456
+local iAP2_BT_out = "iap2bt.out"
+local iAP2_BT_Type = "BLUETOOTH"
+
+-- Device IDs must be the same in order to trigger switching logic
+local iAP2_USB_DeviceID = iAP2_BT_DeviceID
+local iAP2_USB_Port = 34567
+local iAP2_USB_out = "iap2usb.out"
+local iAP2_USB_Type = "USB_IOS"
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+commonSteps:DeletePolicyTable()
+commonSteps:DeleteLogsFiles()
+
+local function createIAP2Device(deviceID, devicePort, deviceOut)
+    local iap2Connection = tcp.Connection(deviceID, devicePort)
+    local fileConnection = file_connection.FileConnection(deviceOut, iap2Connection)
+    local iap2Device = mobile.MobileConnection(fileConnection)
+
+    event_dispatcher:AddConnection(iap2Device)
+
+    return iap2Device
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+local iap2bt_device = 0
+
+Test["Connecting iAP2 Bluetooth"] =
+    function(self)
+    -- iAP2 Bluetooth connection with application registered
+    iap2bt_device = createIAP2Device(iAP2_BT_DeviceID, iAP2_BT_Port, iAP2_BT_out)
+
+    EXPECT_HMICALL(
+        "BasicCommunication.UpdateDeviceList",
+        {
+            deviceList = {{id = config.deviceMAC, name = iAP2_BT_DeviceID, transportType = iAP2_BT_Type}}
+        }
+    ):Do(
+        function(_, data)
+            self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+        end
+    )
+
+    Test:connectMobile(iap2bt_device)
+
+    local iap2bt_mobileSession = mobile_session.MobileSession(self, iap2bt_device, app_RAI_params)
+
+    Test:startSession(iap2bt_mobileSession)
+end
+
+Test["Connecting iAP2 USB"] =
+    function(self)
+    -- iAP2 USB connection with same application
+    local iap2usb_device = createIAP2Device(iAP2_USB_DeviceID, iAP2_USB_Port, iAP2_USB_out)
+
+    EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered"):Times(0)
+    EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered"):Times(0)
+
+    EXPECT_HMICALL(
+        "BasicCommunication.UpdateDeviceList",
+        {
+            -- To check why two devices are in list, maybe an issue
+            deviceList = {
+                {id = config.deviceMAC, name = iAP2_BT_DeviceID, transportType = iAP2_BT_Type},
+                {id = config.deviceMAC, name = iAP2_USB_DeviceID, transportType = iAP2_USB_Type}
+            }
+        }
+    ):Do(
+        function(_, data)
+            self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+
+            -- Disconnect Bluetooth connection
+            iap2bt_device:Close()
+        end
+    )
+
+    Test:connectMobile(iap2usb_device)
+
+    local iap2usb_mobileSession = mobile_session.MobileSession(self, iap2usb_device, app_RAI_params)
+
+    local rpc_service_id = 7
+    iap2usb_mobileSession:StartService(rpc_service_id):Do(
+        function()
+            local correlationId = iap2usb_mobileSession:SendRPC("RegisterAppInterface", app_RAI_params)
```

Comment:
@dev-gh According to the req-t [Main Flow](https://github.com/smartdevicelink/sdl_requirements/blob/feature/iap_over_bt/detailed_docs/iAP_over_BT/iAP_over_BT_re-connection_use_case.md) valid "hashID" value has to be provided within request in order to have correct resumption